git-srep/src/main.rs

```diff
@@ -0,0 +1,176 @@
+use clap::Parser;
+use colored::Colorize;
+use git_utils_shared as git_utils;
+use rayon::prelude::*;
+use std::process::Command;
+
+/// Replace given phrase inside files of the provided paths.
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    /// The word to substitute
+    search: String,
+    /// The word to substitute with
+    replace: String,
+
+    #[clap(
+        long,
+        short,
+        takes_value = false,
+        help = "Substituted word in git dirty files"
+    )]
+    force: bool,
+
+    #[clap(required = true)]
+    paths: Vec<String>,
+}
+
+fn is_file_dirty(file_path: &str) -> bool {
+    let modified = git_utils::file::file_is_modified(file_path);
+    let untracked = git_utils::file::file_is_untracked(file_path);
+    if modified {
+        println!("modified -> {}", file_path.red());
+    }
+    if untracked {
+        println!("Untracked -> {}", file_path.red())
+    }
+
+    modified || untracked
+}
+
+fn change_word_in_files(file_path: Vec<String>, search: &str, replace: &str) {
+    let _ = Command::new("sed")
+        .args([
+            "-i",
+            &format!(
+                "s/{}/{}/g",
+                search.replace("/", r"\/"),
+                replace.replace("/", r"\/")
+            ),
+        ])
+        .args(file_path)
+        .status()
+        .expect("Failed ")
+        .success();
+}
+
+fn main() {
+    let args = Args::parse();
+    // println!("{:?}", args);
+
+    let files: Vec<String> = args
+        .paths
+        .par_iter()
+        .map(|path| {
+            if args.force {
+                git_utils::get_files_with_word_using_grep(&args.search, path)
+            } else {
+                git_utils::get_files_with_word(&args.search, path)
+            }
+        })
+        .filter(|file| file.is_some())
```

Comment:
No need to have 2 functions here.  `git srep` should only ever replace content tracked by git, so you can use `git grep` 100% of the time.

This is a bit of a change from our original discussion because I wasn't thinking that git grep will skip untracked content.  That is the correct behavior though.

Note: even if you wanted to add an option to search and replace untracked content, you can, by using the `--untracked` option to `git grep`.

See detailed notes on the specific options to use when calling `git grep`

Reply:
We won't use this program to replace text in untracked files.